api/v1/serializers/notes-serializer.js

```diff
@@ -0,0 +1,54 @@
+const appRoot = require('app-root-path');
+const _ = require('lodash');
+const JSONAPISerializer = require('jsonapi-serializer').Serializer;
+
+const { serializerOptions } = appRoot.require('utils/jsonapi');
+const { openapi } = appRoot.require('utils/load-openapi');
+const { querySelfLink, idSelfLink } = appRoot.require('utils/uri-builder');
+
+const noteResourceProp = openapi.definitions.NoteResource.properties;
+const noteResourceType = noteResourceProp.type.enum[0];
+const noteResourceKeys = _.keys(noteResourceProp.attributes.properties);
+const noteResourcePath = 'notes';
+
+/**
+ * @summary Serialize noteResources to JSON API
+ * @function
+ * @param {[Object]} rawNotes Raw data rows from data source
+ * @param {Object} query Query parameters
+ * @returns {Object} Serialized noteResources object
+ */
+const serializeNotes = (rawNotes, query) => {
+  const serializerArgs = {
+    identifierField: 'id',
+    resourceKeys: noteResourceKeys,
+    noteResourcePath,
```

Comment:
For each the self-link of each object in `data` array, I got something like `"self": "https://localhost/v1/undefined/111111111-2"`. I think it is due to this line here. In [utils/jsonapi.js](https://github.com/osu-mist/express-api-skeleton/blob/bugs/catch-bugs-and-fix-them/utils/jsonapi.js#L32), the code will try to destructure `serializerArgs` and looking for `resourcePath` to build the datalink URI. The way you set `noteResourcePath` with using the shorthand in `serializerArgs` is actually equivalent to set a key-value pair like `noteResourcePath: noteResourcePath`. To solve the issue, this line should be changed to:

```suggestion
     resourcePath: noteResourcePath,
```

Reply:
Ok, I'll fix this